vignettes/polars.Rmd

```diff
@@ -51,88 +56,105 @@ dependencies other than R itself. This makes it very fast and lightweight to
 # R
-df$select(pl$col(c("A", "B"))$sum())
+df$groupby("id")$mean()
 ```
 
-## Constructors, object types and data types
+## `Series` and `DataFrames`
 
-Let's load the **polars** package and get our feet wet with a tongue-in-cheek
-version of "Hello, world".
+In `polars` objects of class `Series` are analogous to R vectors. Objects of class `DataFrame` are analogous to R data frames. To convert R vectors and data frames to Polars `Series` and `DataFrames`, we load the library and use constructor functions with the `pl$` prefix:
 
 ```{r}
 library(polars)
 
-pl$col("hello")$sum()$over("world", "from")$alias("polars")
-```
+ser = pl$Series((1:5) * 5)
+ser
 
-The key point of this example is the following: All **polars** constructors can
-be accessed via the `pl$` prefix. We'll see this repeatedly in the examples that
-follow.
+dat = pl$DataFrame(mtcars)
+dat
+```
 
-### Data types
 
-Polars is
-[strongly typed](https://en.wikipedia.org/wiki/Strong_and_weak_typing) and new
-types can be created with the `dtypes` constructor. For example:
+Both Polars and R are column-orientated. So you can think of `DataFrames`
+(data.frames) as being made up of a collection of `Series` (vectors). In fact,
+you can create a new Polars `DataFrame` as a mix of `Series` and/or regular R
+vectors.
 
 ```{r}
-# pl$dypes # list all types
-pl$dtypes$Float64
+pl$DataFrame(
+  a = pl$Series((1:5) * 5),
+  b = pl$Series(letters[1:5]),
+  c = pl$Series(c(1, 2, 3, 4, 5)),
+  d = c(15, 14, 13, 12, 11),
+  c(5, 4, 3, 2, 1),
+  1:5
+)
 ```
 
-The full list of valid Polars types can be found by typing `pl$dtypes` 
-into your R console. These include _Boolean_, _Float32(64)_, _Int32(64)_,
-_Utf8_, _Categorical_, _Date_, etc. Note that some type names differ from what
-they are called in R (e.g., _Boolean_ in Polars is equivalent to `logical()` in
-R). This might occasionally require you to look up a specific type. But the good
-news is that **polars** generally does a good job of inferring types
-automatically. We'll see some examples of type coercion later on.
+`Series` and `DataFrame` can be operated on using many standard R functions. For example:
 
-### `Series` and `DataFrames`
+```{r}
+# Series
+length(ser)
 
-The Polars analog to an R vector is a `Series`.
+max(ser)
 
-```{r}
-pl$Series((1:5)*5, name = "my_series")
+# DataFrame
+names(dat)
+
+dim(dat)
+
+head(dat, n = 2)
 ```
 
-Recall that Polars and R are both column-orientated. So you can think of
-`DataFrames` (data.frames) as being made up of a collection of `Series`
-(vectors). In fact, you can create a new Polars `DataFrame` as a mix of `Series`
-and/or regular R vectors.
+## Methods and pipelines
+
+Although some simple R functions work out of the box on **polars** objects, the full power of Polars is realized via _methods_. Polars methods are accessed using the `$` syntax. For example, to convert Polars `Series` and `DataFrames` back to standard R objects, we use the `$to_r_vector()` and `$as_data_frame()` methods:
```

Comment:
It's been a while since I tried this, but is `as_data_frame` still a tibble-exported (or maybe arrow-exported) function? Just worth thinking ahead to whether users might run into a NAMESPACE clash.

Reply:
There's no clash between exported functions and polars methods, though, right? And this PR adds a proper `as.data.frame()` s3 for polars objects, which just calls `df$as_data_frame()` under the hood.

Reply:
Cool cool. Just wanted to check.